### PR TITLE
Update compiler-rt

### PR DESCRIPTION
**Pull request by dschuff**
This change updates emscripten's copy of compiler-rt to trunk r266813.
It also makes the directory layout of emscripten's copy to match
upstream, and uses an archive rather than a linked bitcode file.

Future changes will move the compiler-rt code in library.js to this
archive, and add additional functions.

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -302,7 +302,7 @@ class Dummy:
   system_libs = [('libcxx',      'a',  create_libcxx,      libcxx_symbols,      ['libcxxabi'], True),
                  ('libcxxabi',   'bc', create_libcxxabi,   libcxxabi_symbols,   ['libc'],      False),
                  ('gl',          'bc', create_gl,          gl_symbols,          ['libc'],      False),
-                 ('compiler-rt', 'bc', create_compiler_rt, compiler_rt_symbols, ['libc'],      False)]
+                 ('compiler-rt', 'a', create_compiler_rt, compiler_rt_symbols, ['libc'],      False)]
```
Comment (kripken) [inline tools/system_libs.py]: indent on `['libc'` with the above lines was lost, please restore

Comment (kripken) [inline tools/system_libs.py]: oh actually the indent error begins after the `'a',`

**Comment by kripken** [comment]
Looks good aside from that indentation issue.

For the next PRs, one note is that currently we don't link in compiler-rt for hello world, and I'd like to keep it that way. Unless we measure and it adds 0 time to the build, but I don't think there's any chance of that - it unpacks the archive, links files etc. If compiler-rt is necessary for wasm-backend hello world, maybe we'll diverge asm.js from wasm there, so asm.js will still use the library.js versions of those methods.

File: tools/system_libs.py (hunk 1 above)
Comment (dschuff) [inline tools/system_libs.py]: Done.

**Comment by dschuff** [comment]
I'll check.
But I think just having stuff in library.js that could be linked into the asm module isn't a good idea. Linking from an archive means only those programs that need it pay for it. By putting it in library.js you force all programs to have it, no?

**Comment by kripken** [comment]
No, code in library.js is linked in on a function-by-function basis, so only programs needing it receive it. So it's the same in terms of code size, and it avoids operating on the archive, linking it's contents, and compiling them.

**Comment by dschuff** [comment]
Right, but you have the cost of 'linking in' things from library.js? what does that even mean? how does that work?

**Comment by kripken** [comment]
Linking from JS libraries is essentially free: we have a list of all remaining unresolved symbols, and look them up in a JS object. If found, we just paste that code in the right place in the output. So it's literally just a few fast operations for each unresolved symbol.

**Pull request merged**